src/EventSubscriber/LinkTypePlusEventSubscriber.php

```diff
@@ -63,7 +64,110 @@ public function onlinkTypeCreditCardPaymentEvent(LinkTypePlusEvent $event) {
    */
   public function onlinkTypePayPayEvent(LinkTypePlusEvent $event) {
     $paymentMethod = $event->getPaymentMethod();
-    $this->loggerFactory->notice("$paymentMethod payment event has been subscribed");
-    return TRUE;
+   //Get the payment status and update the payment in Drupal
+   $orderId = $event->getOrderId();
+   $status = $event->getTransitionState();
+   $drupalStatus = $this->webhookStatusMapper($status);
+   $remoteId = $event->getRemoteId();
+   $this->loggerFactory->notice("$paymentMethod payment event has been subscribed");
+   $this->loggerFactory->notice("$orderId \n  $status \n $remoteId");
+   if($this->updatePaymentStatus($orderId, $drupalStatus, $remoteId)){
+     $this->loggerFactory->notice('Status has been updated');
+     return TRUE;
+   }
+  }
+
+
+  /**
+   * Update the status in Drupal.
+   *
+   * @param array $data
+   *   The GMO api response data.
+   */
+  public function updatePaymentStatus($order_id, $status, $remote_id){
+    if ($order_id && !empty($order_id)) {
+      $order = Order::load($order_id);
+      $payment_storage = $this->entityTypeManager->getStorage('commerce_payment');
+      $paymentGateway = $order->get('payment_gateway')->entity->id();
+      $total_price = $order->getTotalprice()->getNumber();
+      $currency = $order->getTotalprice()->getCurrencyCode();
+      $payment = $payment_storage->loadByProperties([
+        'order_id' => $order_id,
+      ]);
+      if ($payment) {
+        $payment = array_shift($payment);
+        $payment->setState($status);
+        $payment->setRemoteId($remote_id);
+        $payment->save();
+      } else {
+        $payment = $payment_storage->create([
+          'state' => $status,
+          'payment_gateway' => $paymentGateway,
+          'remote_id' => $remote_id,
+          'amount' => [
+            'number' => $total_price,
+            'currency_code' => $currency,
+          ],
+          'order_id' => $order_id,
+        ]);
+        $payment->save();
+      }
+      return TRUE;
+    }
+  }
+
+  /**
+   * Method used to track the linktype status with the drupal commerce_payment.
+   *
+   * We got these status through webhook.
+   *
+   * It may differs for credit card, paypay etc.
+   * Please add the required status by refering the doc
+   *
+   * Refer: https://docs.mul-pay.jp/payment/credit/notice
+   *        https://docs.mul-pay.jp/paypay/payg-notice
+   *        https://docs.mul-pay.jp/
+   *
+   * THIS SECTION NEEDS TO BE EXTENDED.
```

Comment:
webhookStatus can be changed to creditCardStatusMapper() - we can map the credit card status alone for paypal they can separate mapping. 

check the status of the credit card in this link - https://docs.mul-pay.jp/payment/credit/notice